sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -123,6 +126,24 @@ private[hive] object HiveShim {
     // for Serialization
     def this() = this(null)
 
+    override def hashCode(): Int = {
+        if (instance != null && instance.isInstanceOf[GenericUDFMacro]) {
```

Comment:
Hive's macro is kind of special. I do not think users can use a class that extends GenericUDFMacro (You can take a look at hive's function registry. Macro registration is handled by registerMacro.).